doc/toolchains/llvm.chapter.md

```diff
@@ -0,0 +1,19 @@
+# The LLVM Toolchain {#chap-toolchains}
+
+LLVM is a target-independent optimizer and code generator and serves as the basis for many languages such as Haskell, Rust, Zig, and many other languages. It forms the base tools for Apple's Darwin platform.
+
+## Using LLVM {#sec-using-llvm}
+
+LLVM has two ways of being used. One is by using it across all of nixpkgs and the other is to compile and build other packages.
+
+### Building packages with LLVM {#sec-building-packages-with-llvm}
+
+To build a package with LLVM, the simplest way is to invoke it with `pkgsLLVM`. The simplest way is to run:
```

Comment:
Could we explain that pkgsLLVM is equivalent to importing Nixpkgs a certain way? I think it would be really good if people knew that `pkgsCross`, `pkgsLLVM` etc. aren't magic.

Reply:
Yes, I think that'd make sense. I can also show how you can setup LLVM cross compiling via `crossPlatform` when importing nixpkgs.

Reply:
Done

Reply:
Could we order it the other way around? Explain what's happening, and then introduce pkgsLLvm as a shorthand?

Reply:
Ok, it's more verbose as to how things function.